src/Avalonia.Visuals/Utility/ReadOnlySlice.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Avalonia.Utility
+{
+    public readonly struct ReadOnlySlice<T> : IReadOnlyList<T>
```

Comment:
Could you explain why this is needed? Why can't we use `ReadOnlyMemory<T>.Slice()`?

Reply:
A slice can't tell you anything about the source. We need to know where we are within a sequence of glyphs for example.